src/controller/CHIPCluster.cpp

```diff
@@ -42,10 +42,11 @@ CHIP_ERROR ClusterBase::Associate(DeviceProxy * device, EndpointId endpoint)
     return err;
 }
 
+/**
+ * To be used in a test context where a SecureSession is already present
+ */
 CHIP_ERROR ClusterBase::AssociateWithGroup(DeviceProxy * device, GroupId groupId)
```

Comment:
So this function cannot be used outside of a YAML context? What's the plan to enable this in chip-tool?

If I'm reading the current logic correctly, it requires `mDevice` to always have a valid secure session ahead of time to permit using it in a group setting, which is an odd pre-requisite.

This is all because mDevice is starting out pointing to a unicast destination - why can't it start out pointing to a groupcast destination instead?